addons/portal/wizard/portal_share.py

```diff
@@ -44,16 +44,19 @@ def _compute_access_warning(self):
                     record = res_model.browse(rec.res_id)
                     rec.access_warning = record.access_warning
 
-    def action_send_mail(self):
-        active_record = self.env[self.res_model].browse(self.res_id)
-        note = self.env.ref('mail.mt_note')
-        signup_enabled = self.env['ir.config_parameter'].sudo().get_param('auth_signup.invitation_scope') == 'b2c'
+    @api.model
+    def _get_note(self):
+        return self.env.ref('mail.mt_note')
 
-        if hasattr(active_record, 'access_token') and active_record.access_token or not signup_enabled:
-            partner_ids = self.partner_ids
-        else:
-            partner_ids = self.partner_ids.filtered(lambda x: x.user_ids)
-        # if partner already user or record has access token send common link in batch to all user
+    def _get_active_record(self):
+        self.ensure_one()
+        return self.env[self.res_model].browse(self.res_id)
+
+    def _send_public_link(self, active_record, note, partners=None):
+        if not note:
+            note = self._get_note()
```

Comment:
What is the point of these fallbacks, or even this parameter? These methods are used once, with a `note` parameter provided, with a hardcoded value.

In fact given the (somewhat ill) naming this has no reason to be overridden at all: it's not any sort of note, instead it's named after being the note subtype. If the subtype can be replaced then it's not `note` anymore is it? And I doubt this is going to be called so often we need to memoize a single extra ref call, given both methods are internally batched.

Reply:
I split the content in the `action_send_mail` method to easily use what I need in `project.share.wizard` since this new wizard inherits to `portal.share`.
So, if I does not pass the note because I don't have it in `action_send_mail` method in `project.share.wizard` for instance then the note will always be initialized thanks to this check and `_get_note` method.  Normally, no extra ref call should be done. 🤔 
But indeed, I could get the note in the `action_send_mail`, pass it as argument in the method (like I did in `portal.share`) and make the `note` parameter mandatory instead of optional in both methods.

Reply:
> So, if I does not pass the note because I don't have it in action_send_mail method in project.share.wizard for instance then the note will always be initialized thanks to this check and _get_note method. Normally, no extra ref call should be done. thinking

What I'm saying is the extra ref call probably doesn't matter, having this be a parameter is confusing when it is essentially a constant: what use case is there for using anything other than the note subtype?